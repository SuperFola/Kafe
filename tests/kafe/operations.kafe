x: int = 1 + 1
x: int = 2
x: int = 2 * 4 - 5 + x
fun main() -> int
    x : int = 10
end