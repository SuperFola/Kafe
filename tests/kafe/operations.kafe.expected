(Program
    (Definition
        (VarName x)
        (Type int)
        (OperationsList
            (Integer 1)
            (Operator +)
            (Integer 1)
        )
    )
    (Definition
        (VarName x)
        (Type int)
        (Integer 2)
    )
    (Definition
        (VarName x)
        (Type int)
        (OperationsList
            (Integer 2)
            (Operator *)
            (Integer 4)
            (Operator -)
            (Integer 5)
            (Operator +)
            (VarUse x)
        )
    )
    (Function
        (Name main)
        (Args)
        (Type int)
        (Body
            (Definition
                (VarName x)
                (Type int)
                (Integer 10)
            )
        )
    )
)